packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -435,6 +435,9 @@ const QueryEditorHeaderContent = observer(
 
     const handleQuerySaveAs = (): void => {
       editorStore.queryCreatorState.setShowCreateModal(true);
+      if (editorStore instanceof ExistingQueryEditorStore) {
```

Comment:
why don't we set this when we initialize the state for `ExistingQueryEditorStore` instead of doing this here every time user hits the `Save As` button?

Reply:
Let's be slightly cleaner, let's create 2 methods: `openModal` and `close`

```ts
open(originalQuery?: Query | undefined)
close()
```